### ISSUE TITLE
Can't compile SciPy using gcc and OpenBLAS

### ISSUE THREAD

**Issue by Juanlu001** [ContinuumIO/anaconda-issues#10495]
### Actual Behavior

The compilation fails with this message:

```
    error: Command "gcc -pthread -shared -Wl,-rpath,/home/juanlu/.miniconda36/envs/scipy37/lib -L/home/juanlu/.miniconda36/envs/scipy37/lib -fuse-linker-plugin -ffat-lto-objects -flto-partition=none -g -Wl,-rpath,/home/juanlu/.miniconda36/envs/scipy37/lib -L/home/juanlu/.miniconda36/envs/scipy37/lib build/temp.linux-x86_64-3.7/scipy/cluster/_vq.o -L/home/juanlu/.miniconda36/envs/scipy37/lib -Lbuild/temp.linux-x86_64-3.7 -lopenblas -lopenblas -o scipy/cluster/_vq.cpython-37m-x86_64-linux-gnu.so -Wl,--version-script=build/temp.linux-x86_64-3.7/link-version-scipy.cluster._vq.map" failed with exit status 1
    gcc: error: -fuse-linker-plugin is not supported in this configuration
    gcc: error: -fuse-linker-plugin is not supported in this configuration
Cleaning up...
```

### Expected Behavior

Being able to compile SciPy.

### Steps to Reproduce

Environment creation:

<details>

```
$ conda create -n scipy37 python=3.7 defaults::gcc defaults::openblas -yq && conda activate scipy37 && pip install cython tempita pytest ipython && pip install -e . -vv > build.log 2>&1
Solving environment: ...working... done

## Package Plan ##

  environment location: /home/juanlu/.miniconda36/envs/scipy37

  added / updated specs: 
    - defaults::gcc
    - defaults::openblas
    - python=3.7


The following NEW packages will be INSTALLED:

    blas:            1.0-openblas          conda-forge
    bzip2:           1.0.6-h470a237_2      conda-forge
    ca-certificates: 2018.11.29-ha4d7672_0 conda-forge
    certifi:         2018.11.29-py37_1000  conda-forge
    cloog:           0.18.0-0                         
    gcc:             4.8.5-7                          
    gmp:             6.1.2-hfc679d8_0      conda-forge
    isl:             0.12.2-0                         
    libffi:          3.2.1-hfc679d8_5      conda-forge
    libgcc-ng:       8.2.0-hdf63c60_1                 
    libgfortran-ng:  7.2.0-hdf63c60_3      conda-forge
    libopenblas:     0.3.3-h5a2b251_3                 
    libstdcxx-ng:    7.2.0-hdf63c60_3      conda-forge
    mpc:             1.1.0-4               conda-forge
    mpfr:            3.1.5-0               conda-forge
    ncurses:         6.1-hfc679d8_2        conda-forge
    nomkl:           3.0-0                            
    openblas:        0.3.3-3                          
    openblas-devel:  0.3.3-3                          
    openssl:         1.0.2p-h470a237_2     conda-forge
    pip:             18.1-py37_1000        conda-forge
    python:          3.7.1-h5001a0f_0      conda-forge
    readline:        7.0-haf1bffa_1        conda-forge
    setuptools:      40.6.3-py37_0         conda-forge
    sqlite:          3.26.0-hb1c47c0_0     conda-forge
    tk:              8.6.9-ha92aebf_0      conda-forge
    wheel:           0.32.3-py37_0         conda-forge
    xz:              5.2.4-h470a237_1      conda-forge
    zlib:            1.2.11-h470a237_4     conda-forge

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
Collecting cython
  Using cached https://files.pythonhosted.org/packages/25/63/14ded19bc584405b7633a3ba8e2019f2126f214eca2a88b79b2d71e6aae4/Cython-0.29.2-cp37-cp37m-manylinux1_x86_64.whl
Collecting tempita
Collecting pytest
  Using cached https://files.pythonhosted.org/packages/95/6a/4122affe57eb3857874ef959a73f362e51b23812b96c92e073b16e1effd0/pytest-4.1.0-py2.py3-none-any.whl
Collecting ipython
  Using cached https://files.pythonhosted.org/packages/f0/b4/a9ea018c73a84ee6280b2e94a1a6af8d63e45903eac2da0640fa63bca4db/ipython-7.2.0-py3-none-any.whl
Collecting six>=1.10.0 (from pytest)
  Using cached https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting py>=1.5.0 (from pytest)
  Using cached https://files.pythonhosted.org/packages/3e/c7/3da685ef117d42ac8d71af525208759742dd235f8094221fdaafcd3dba8f/py-1.7.0-py2.py3-none-any.whl
Collecting attrs>=17.4.0 (from pytest)
  Using cached https://files.pythonhosted.org/packages/3a/e1/5f9023cc983f1a628a8c2fd051ad19e76ff7b142a0faf329336f9a62a514/attrs-18.2.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools in /home/juanlu/.miniconda36/envs/scipy37/lib/python3.7/site-packages (from pytest) (40.6.3)
Collecting more-itertools>=4.0.0 (from pytest)
  Using cached https://files.pythonhosted.org/packages/a4/a6/42f17d065bda1fac255db13afc94c93dbfb64393eae37c749b4cb0752fc7/more_itertools-5.0.0-py3-none-any.whl
Collecting pluggy>=0.7 (from pytest)
  Using cached https://files.pythonhosted.org/packages/1c/e7/017c262070af41fe251401cb0d0e1b7c38f656da634cd0c15604f1f30864/pluggy-0.8.0-py2.py3-none-any.whl
Collecting atomicwrites>=1.0 (from pytest)
  Using cached https://files.pythonhosted.org/packages/3a/9a/9d878f8d885706e2530402de6417141129a943802c084238914fa6798d97/atomicwrites-1.2.1-py2.py3-none-any.whl
Collecting pexpect; sys_platform != "win32" (from ipython)
  Using cached https://files.pythonhosted.org/packages/89/e6/b5a1de8b0cc4e07ca1b305a4fcc3f9806025c1b651ea302646341222f88b/pexpect-4.6.0-py2.py3-none-any.whl
Collecting jedi>=0.10 (from ipython)
  Using cached https://files.pythonhosted.org/packages/c2/bc/54d53f5bc4658380d0eca9055d72be4df45e5bfd91a4bac97da224a92553/jedi-0.13.2-py2.py3-none-any.whl
Collecting pickleshare (from ipython)
  Using cached https://files.pythonhosted.org/packages/9a/41/220f49aaea88bc6fa6cba8d05ecf24676326156c23b991e80b3f2fc24c77/pickleshare-0.7.5-py2.py3-none-any.whl
Collecting prompt-toolkit<2.1.0,>=2.0.0 (from ipython)
  Using cached https://files.pythonhosted.org/packages/d1/e6/adb3be5576f5d27c6faa33f1e9fea8fe5dbd9351db12148de948507e352c/prompt_toolkit-2.0.7-py3-none-any.whl
Collecting traitlets>=4.2 (from ipython)
  Using cached https://files.pythonhosted.org/packages/93/d6/abcb22de61d78e2fc3959c964628a5771e47e7cc60d53e9342e21ed6cc9a/traitlets-4.3.2-py2.py3-none-any.whl
Collecting pygments (from ipython)
  Using cached https://files.pythonhosted.org/packages/13/e5/6d710c9cf96c31ac82657bcfb441df328b22df8564d58d0c4cd62612674c/Pygments-2.3.1-py2.py3-none-any.whl
Collecting decorator (from ipython)
  Using cached https://files.pythonhosted.org/packages/bc/bb/a24838832ba35baf52f32ab1a49b906b5f82fb7c76b2f6a7e35e140bac30/decorator-4.3.0-py2.py3-none-any.whl
Collecting backcall (from ipython)
Collecting ptyprocess>=0.5 (from pexpect; sys_platform != "win32"->ipython)
  Using cached https://files.pythonhosted.org/packages/d1/29/605c2cc68a9992d18dada28206eeada56ea4bd07a239669da41674648b6f/ptyprocess-0.6.0-py2.py3-none-any.whl
Collecting parso>=0.3.0 (from jedi>=0.10->ipython)
  Using cached https://files.pythonhosted.org/packages/09/51/9c48a46334be50c13d25a3afe55fa05c445699304c5ad32619de953a2305/parso-0.3.1-py2.py3-none-any.whl
Collecting wcwidth (from prompt-toolkit<2.1.0,>=2.0.0->ipython)
  Using cached https://files.pythonhosted.org/packages/7e/9f/526a6947247599b084ee5232e4f9190a38f398d7300d866af3ab571a5bfe/wcwidth-0.1.7-py2.py3-none-any.whl
Collecting ipython-genutils (from traitlets>=4.2->ipython)
  Using cached https://files.pythonhosted.org/packages/fa/bc/9bd3b5c2b4774d5f33b2d544f1460be9df7df2fe42f352135381c347c69a/ipython_genutils-0.2.0-py2.py3-none-any.whl
Installing collected packages: cython, tempita, six, py, attrs, more-itertools, pluggy, atomicwrites, pytest, ptyprocess, pexpect, parso, jedi, pickleshare, wcwidth, prompt-toolkit, decorator, ipython-genutils, traitlets, pygments, backcall, ipython
Successfully installed atomicwrites-1.2.1 attrs-18.2.0 backcall-0.1.0 cython-0.29.2 decorator-4.3.0 ipython-7.2.0 ipython-genutils-0.2.0 jedi-0.13.2 more-itertools-5.0.0 parso-0.3.1 pexpect-4.6.0 pickleshare-0.7.5 pluggy-0.8.0 prompt-toolkit-2.0.7 ptyprocess-0.6.0 py-1.7.0 pygments-2.3.1 pytest-4.1.0 six-1.12.0 tempita-0.5.2 traitlets-4.3.2 wcwidth-0.1.7
```
</details>

Build log: https://gist.github.com/Juanlu001/94a3fcfb9192901d21c685405ceb6a6d

##### Operating System:

Linux Mint 19.1.

##### `conda info`

```

     active environment : None
            shell level : 0
       user config file : /home/juanlu/.condarc
 populated config files : /home/juanlu/.miniconda36/.condarc
                          /home/juanlu/.condarc
          conda version : 4.5.12
    conda-build version : 3.17.5
         python version : 3.6.7.final.0
       base environment : /home/juanlu/.miniconda36  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/linux-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /home/juanlu/.miniconda36/pkgs
                          /home/juanlu/.conda/pkgs
       envs directories : /home/juanlu/.miniconda36/envs
                          /home/juanlu/.conda/envs
               platform : linux-64
             user-agent : conda/4.5.12 requests/2.21.0 CPython/3.6.7 Linux/4.15.0-43-generic linuxmint/19.1 glibc/2.27
                UID:GID : 1000:1000
             netrc file : /home/juanlu/.netrc
           offline mode : False


```

##### `conda list --show-channel-urls`

```
# packages in environment at /home/juanlu/.miniconda36:
#
# Name                    Version                   Build  Channel
asn1crypto                0.24.0                py36_1003    conda-forge
attrs                     18.2.0                     py_0    conda-forge
beautifulsoup4            4.6.3                 py36_1000    conda-forge
blinker                   1.4                        py_1    conda-forge
bzip2                     1.0.6                h470a237_2    conda-forge
ca-certificates           2018.11.29           ha4d7672_0    conda-forge
certifi                   2018.11.29            py36_1000    conda-forge
cffi                      1.11.5           py36h5e8e0c9_1    conda-forge
chardet                   3.0.4                 py36_1003    conda-forge
click                     7.0                        py_0    conda-forge
conda                     4.5.12                py36_1000    conda-forge
conda-build               3.17.5                   py36_0    conda-forge
conda-env                 2.6.0                         1    conda-forge
conda-forge-pinning       2018.12.18                    0    conda-forge
conda-smithy              3.2.2                      py_2    conda-forge
conda-verify              3.1.1                 py36_1000    conda-forge
cryptography              2.2.1            py36hdffb7b8_1    conda-forge
cryptography-vectors      2.4.2                   py_1000    conda-forge
expat                     2.2.5                hfc679d8_2    conda-forge
filelock                  3.0.10                     py_0    conda-forge
future                    0.17.0                py36_1000    conda-forge
gitdb2                    2.0.5                      py_0    conda-forge
gitpython                 2.1.11                     py_0    conda-forge
glob2                     0.6                        py_0    conda-forge
idna                      2.8                   py36_1000    conda-forge
isodate                   0.6.0                      py_1    conda-forge
jinja2                    2.10                       py_1    conda-forge
libarchive                3.3.3                h823be47_0    conda-forge
libedit                   3.1.20170329         haf1bffa_1    conda-forge
libffi                    3.2.1                hfc679d8_5    conda-forge
libgcc-ng                 7.2.0                hdf63c60_3    conda-forge
libiconv                  1.15                 h470a237_3    conda-forge
liblief                   0.9.0                h1532aa0_0    defaults
libstdcxx-ng              7.2.0                hdf63c60_3    conda-forge
markupsafe                1.1.0            py36h470a237_0    conda-forge
msrest                    0.6.2                      py_0    conda-forge
ncurses                   6.1                  hfc679d8_2    conda-forge
oauthlib                  2.1.0                      py_0    conda-forge
openssl                   1.0.2p               h470a237_1    conda-forge
patchelf                  0.9                  hfc679d8_2    conda-forge
pip                       18.1                  py36_1000    conda-forge
pkginfo                   1.4.2                      py_1    conda-forge
psutil                    5.4.8            py36h470a237_0    conda-forge
py-lief                   0.9.0            py36h1532aa0_0    defaults
pycosat                   0.6.3            py36h470a237_1    conda-forge
pycparser                 2.19                       py_0    conda-forge
pycrypto                  2.6.1            py36h470a237_2    conda-forge
pygithub                  1.39                  py36_1000    conda-forge
pyjwt                     1.6.4                      py_0    conda-forge
pyopenssl                 18.0.0                py36_1000    conda-forge
pysocks                   1.6.8                 py36_1002    conda-forge
python                    3.6.7                h5001a0f_1    conda-forge
python-libarchive-c       2.8                   py36_1004    conda-forge
pytz                      2018.7                     py_0    conda-forge
pyyaml                    3.13             py36h470a237_1    conda-forge
readline                  7.0                  haf1bffa_1    conda-forge
requests                  2.21.0                py36_1000    conda-forge
requests-oauthlib         1.0.0                      py_1    conda-forge
ruamel.yaml               0.15.81          py36h470a237_0    conda-forge
ruamel_yaml               0.15.71          py36h470a237_0    conda-forge
setuptools                40.6.3                   py36_0    conda-forge
six                       1.12.0                py36_1000    conda-forge
smmap2                    2.0.5                      py_0    conda-forge
sqlite                    3.26.0               hb1c47c0_0    conda-forge
tk                        8.6.9                ha92aebf_0    conda-forge
tqdm                      4.28.1                     py_0    conda-forge
urllib3                   1.24.1                py36_1000    conda-forge
vsts-python-api           0.1.21                     py_0    conda-forge
wheel                     0.32.3                   py36_0    conda-forge
xz                        5.2.4                h470a237_1    conda-forge
yaml                      0.1.7                h470a237_1    conda-forge
zlib                      1.2.11               h470a237_3    conda-forge

```

**Comment by msarahan** [comment]
Please refer to our recipe at https://github.com/anacondarecipes/scipy-feedstock

The gcc is package is deprecated and definitely not a supported way to compile packages.  You can install the compiler activation scripts that conda-build uses instead, but make sure you activate the environment with the compilers (activation sets the env vars that establish how to build things).

**Comment by Juanlu001** [comment]
> Please refer to our recipe at https://github.com/anacondarecipes/scipy-feedstock

Thanks for the suggestion. I am aware I can install SciPy with conda, but that doesn't really address the issue.

> The gcc is package is deprecated and definitely not a supported way to compile packages.

I tried because @rgommers suggested that [a few days ago](https://github.com/scipy/scipy/issues/8325#issuecomment-450593245). Is this deprecation noted somewhere?

**Comment by msarahan** [comment]
I pointed you at the recipe as an example of building it from source. I didn't tell you to conda install scipy.

The debug feature of conda build may be helpful to you: https://github.com/conda/conda-build/blob/master/docs/source/debugging.rst

**Comment by mingwandroid** [comment]
> Is this deprecation noted somewhere?

The `gcc` package was a stop-gap measure so that *we* could build software that needed more modern compilers. I am not aware of us recommending anyone to use them officially (except perhaps in rare cases where they wanted to build something using our recipes) so I don't *really* see the need to tell people to stop using them now.

We fully endorse people using our new compilers though: https://conda.io/docs/user-guide/tasks/build-packages/compiler-tools.html and https://www.anaconda.com/blog/developer-blog/utilizing-the-new-compilers-in-anaconda-distribution-5/

**Comment by Juanlu001** [comment]
Thanks @rgommers!

**Comment by jjhelmus** [comment]
I touched on this in scipy/scipy#8325. I issue is that the mkl/openblas libraries and `libgfortran.so` are both in the anaconda_env/lib directory.  The presence of the mkl/openblas libraries add this path to the linker's library search path prior to any system directory (like /usr/lib/gcc/x86_64-linux-gnu/5).  The libgfortran.so library in the prefix/lib directory is linked but may be incompatible with the libgfortran version the objects were compiled for.  

I was able to build the development version of SciPy by either renaming this file (add .hidden, etc) or by installing the `gcc_linux-64`, `gxx_linux-64` and `gfortran_linux-64` packages.  The later required that the environment be activated and `-shared` be added to LDFLAGS using for example `export LDFLAGS="$LDFLAGS -shared"`
